Добавлено задание 8.3

POSTDAR opened the pull request.

Commit:
Добавлено задание 8.3

roman-rtacademy commented on 8/8.3/8.3_mariadb.sql:
```diff
@@ -0,0 +1,17 @@
+    timezone          varchar(37)                           not null,
+    registration_time timestamp current_timestamp()         not null,
+    registration_ip   varchar(15)                           not null,
+    UNIQUE (id),
```
Воно вже PK, можна не писати Unique

roman-rtacademy commented on 8/8.3/8.3_mariadb.sql:
```diff
@@ -0,0 +1,17 @@
+    registration_time timestamp current_timestamp()         not null,
+    registration_ip   varchar(15)                           not null,
+    UNIQUE (id),
+    UNIQUE (email),
```
Тобто я можу вставити 2 email: `roman@rtacademy.net` та `rOmAn@rtacademy.net` ?

POSTDAR replied:
Чому 2? якщо стоїть unique то я не зможу вписати дві такі адреси..
Так?

roman-rtacademy replied:
Різний регістр - різні коди ASCII, отже - різні символи.

POSTDAR replied:
Ну тоді я так розумію що unique спрацює. Якщо різний регістр це різні символи.

roman-rtacademy replied:
Спробуй використати функцію lower()

POSTDAR replied:
Я читаю і lover можна зробити тільки вибіркою select
Але це ж вже постфактум 
Мені здається це не правильно.

roman-rtacademy replied:
Різний регістр це різні символи - отже в тебе у таблиці будуть дублі

roman-rtacademy replied:
Додай `UNIQUE(lower(email))` та прибери `CHECK ( LOWER(email) )` та `UNIQUE (email)`.
Це і буде відповідь.

roman-rtacademy commented on 8/8.3/8.3_mariadb.sql:
```diff
@@ -0,0 +1,17 @@
+    UNIQUE (id),
+    UNIQUE (email),
+    PRIMARY KEY (id),
+    CHECK ( phone = 13)
```
Телефон може бути лише цифрою `13` ?

POSTDAR replied:
А якщо так?
CHECK ( phone = '+^[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]*')

roman-rtacademy replied:
Спробувала щось вставити? Працює?

POSTDAR replied:
Не працює. Тоді що можна зробити?

POSTDAR replied:
Обмежити мінімальну довжину?

roman-rtacademy replied:
Напиши правильно CHECK на регулярний вираз

https://dev.mysql.com/doc/refman/8.0/en/regexp.html

POSTDAR replied:
CHECK ( phone ='/+^[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]*/')
Так правильно?

roman-rtacademy replied:
Перевіряла? А чому останнє число "*" ?

roman-rtacademy commented on 8/8.3/8.3_postrgesql.sql:
```diff
@@ -0,0 +1,13 @@
+                       timezone char(37)            NOT NULL ,
+                       registration_time timestamp  NOT NULL ,
+                       registration_ip char(15)     NOT NULL
+)
```
Теж саме, що і для mysql

roman-rtacademy reviewed (changes requested):
Див. коментарі

roman-rtacademy reviewed (changes requested):
Відповіді у коментарях

Commit:
Виправлено завдання 8,3

Commit:
Исправленно задание 8.3
Добавлена проверка чтобы email был lower, а потом уникальный.
По другому я не могу вставить LOWER -ругается постоянно

Commit:
Добавлено задание 8.7

Commit:
Исправленно задание 8.3
Добавлена проверка чтобы email был lower, а потом уникальный.
По другому я не могу вставить LOWER -ругается постоянно

roman-rtacademy reviewed (changes requested):
Нагадую про мої коментарі до цього PR.

Commit:
Revert "Исправленно задание 8.3"

This reverts commit 55d828856f5ae519af9d57a8e6f53fb51d9ad425.

Commit:
Revert "Добавлено задание 8.7"

This reverts commit a4164639cefd142ae29a91db588cca140fe0626a.

Commit:
Merge branch '8.3' of github.com:POSTDAR/rtacademy-homework into 8.3

roman-rtacademy reviewed (changes requested):
Див. коментар
